### ISSUE TITLE
Supporting platform accessibility actions for custom keyboards

### ISSUE THREAD

**Issue by jitinsharma** [flutter/flutter#90498]
## Use case
I'm trying to implement a custom keyboard and want to match accessibility actions of framework keyboards of Android/iOS but am unable to do so.

I'm using following `Semantics` for keys of custom keyboard.
```dart
Semantics.fromProperties(
  properties: SemanticsProperties(
    label: "some-semantics-label",
    keyboardKey: true,
)
```

### Expected Behaviour

**Android**
When Talkback is on, Android system keyboard doesn't use double tap to input the key instead when you tap on the key, its semantic label is pronounced by Talkback and the key is entered into the input field. On Flutter using `keyboardKey: true` on our custom widget does not reproduce similar behaviour and I'm unable to find a way to remove double tap behaviour on Android. 

Here's a video which demonstrates android's system keyboard accessbiility: https://www.youtube.com/watch?v=y2tT4jDRfWs

On Native Android, we achieve this using following piece of code
```kotlin
ViewCompat.setAccessibilityDelegate(view, object : AccessibilityDelegateCompat() {
    override fun onInitializeAccessibilityNodeInfo(host: View, info: AccessibilityNodeInfoCompat) {
        super.onInitializeAccessibilityNodeInfo(host, info)
        info.removeAction(AccessibilityNodeInfoCompat.AccessibilityActionCompat.ACTION_CLICK)
        info.isClickable = false
    }
})
```

**iOS**
On iOS, the default behaviour of system keyboard, when voice over is on, is to pronounce the key semantic label but activate it only on double tap. However this behaviour can be overriden in voice over settings by turning on a setting called `touch typing`, after which the key can be input using a single tap. Again using `keyboardKey: true` on our custom widget does not reproduce similar behaviour instead it always requires double tap to input key irrespective of voice over touch typing setting.

Here's a video which demonstrates various modes of iOS's system keyboard with accessibility: https://www.youtube.com/watch?v=1O2hwY-mGyQ

On Native iOS, we achieve this by using `keyboardKey` accessibility trait
https://developer.apple.com/documentation/uikit/uiaccessibility/uiaccessibilitytraits/1620182-keyboardkey

**Note**
Full sample code: https://github.com/jitinsharma/flutter_custom_keyboard/blob/master/lib/main.dart#L115

## Proposal
The expectation is `keyboardKey: true` should follow accessibility actions of framework keyboards on each Platform, which would help in replicating these actions on any flutter widget.

**Comment by zanderso** [comment]
@chunhtai FYI This might just be a matter of plumbing the trait up to the framework.

**Comment by chunhtai** [comment]
This trait was originally added for fuchsia use case and are not used by any other platform
It looks like in iOS we should set the accessibilityTrait to https://developer.apple.com/documentation/uikit/uiaccessibilitytraitkeyboardkey?language=objc

in Android
we need to set
https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#setTextEntryKey(boolean)
